Wikipedia/UI-V5/WMFArticleFetcher.m

```diff
@@ -20,37 +188,45 @@ @implementation WMFArticleFetcher
 - (instancetype)initWithDataStore:(MWKDataStore*)dataStore {
     self = [super init];
     if (self) {
-        self.dataStore = dataStore;
-        AFHTTPRequestOperationManager* manager = [AFHTTPRequestOperationManager wmf_createDefaultManager];
-        manager.responseSerializer = [AFHTTPResponseSerializer serializer];
-        self.operationManager      = manager;
+        self.operationManager.requestSerializer  = [WMFArticleRequestSerializer serializer];
+        self.operationManager.responseSerializer = [WMFArticleResponseSerializer serializer];
+        self.dataStore                           = dataStore;
     }
     return self;
 }
 
-- (ArticleFetcher*)fetcher {
-    if (!_fetcher) {
-        _fetcher = [[ArticleFetcher alloc] init];
-    }
+- (id)serializedArticleWithTitle:(MWKTitle*)title response:(NSDictionary*)response {
+    MWKArticle* article = [self.dataStore articleWithTitle:title];
```

Comment:
I didn't change the serialization logic, just moved it. 

Having said that… I think you are right that we do want to totally overwrite any existing article. However, since the article is not immutable, there may be logic that makes changes to the article elsewhere in the app. I am not sure that we can say it is safe to overwrite what is in the store until we make it immutable. 

It also may be "legacy" logic from when it was core data and you would always do a fetch before writing to make sure you don't have duplicates in the database.